app/src/main/java/com/example/simplecaclulator/CalculatorStringParser.kt

```diff
@@ -0,0 +1,155 @@
+                            index
+                        )
+                    ) - evaluate(operandsAndOperators.subList(index + 1, operandsAndOperators.size))
+
+                } else {
+                    operandsAndOperators[index + 1] =
+                        operandsAndOperators[index] + operandsAndOperators[index + 1]
+
+                    return@evaluate evaluate(
+                        (operandsAndOperators.subList(
+                            0,
+                            index
+                        ) + operandsAndOperators.subList(
+                            index + 1,
+                            operandsAndOperators.size
+                        )) as MutableList<String>
+                    )
+                }
+            }
+        }
+        forEachIndexed { index, s ->
+            if (s == Operands.PlUS.sign.toString()) {
+                return@evaluate evaluate(
+                    operandsAndOperators.subList(
+                        0,
+                        index
+                    )
+                ) + evaluate(operandsAndOperators.subList(index + 1, operandsAndOperators.size))
+            }
+        }
+        forEachIndexed { index, s ->
+            if (s == Operands.MULTIPLY.sign.toString()) {
+                return@evaluate evaluate(
+                    operandsAndOperators.subList(
+                        0,
+                        index
+                    )
+                ) * evaluate(operandsAndOperators.subList(index + 1, operandsAndOperators.size))
+            }
+        }
+        forEachIndexed { index, s ->
+            if (s == Operands.DIVISION.sign.toString()) {
+                return@evaluate evaluate(
+                    operandsAndOperators.subList(
+                        0,
+                        index
+                    )
+                ) / evaluate(operandsAndOperators.subList(index + 1, operandsAndOperators.size))
+            }
+        }
+    }
+    if (operandsAndOperators.size == 1) {
+        return operandsAndOperators[0].toDouble()
+    }
+    throw IllegalArgumentException()
+}
+
+private fun checkNext(input: String, id: Int): Boolean {
+    return if (id < input.lastIndex) {
+        input[id + 1] in '0'..'9' || input[id + 1] == Operands.MINUS.sign
+    } else false
+}
+
+private fun String.parsePercents(): MutableList<String> {
+    val numbers = mutableListOf<String>()
+    this.splitByOperators().forEach {
+        val firstPercentId = it.indexOf(Operands.PERCENT.sign)
+        numbers.add(
+            if (firstPercentId > 0) {
+                (it
+                    .substring(
+                        0, firstPercentId
+                    ).toDouble() / (100.0.pow((it.length - firstPercentId).toDouble())
+                        )).toString()
+            } else
+                it
+        )
+    }
+    Log.d(TAG, "parsePercents: $numbers")
+    return numbers
+}
+
+private fun isOperand(char: Char): Boolean {
+    Operands.values()
```

Comment:
method is isOperand works for all cases but PERCENT. Maybe PERCENT isn't operand?